src/ruby/puppet-server-lib/puppet/server/config.rb

```diff
@@ -25,6 +25,15 @@ def self.initialize_puppet_server(puppet_server_config)
       Puppet::Util::Profiler.add_profiler(@profiler)
     end
 
+    if puppet_server_config["shared_terminus_state"]
+      begin
+        require 'puppet/util/puppetdb/shared_terminus_state'
+        Puppet::Util::Puppetdb::SharedTerminusState.shared_terminus_state = puppet_server_config["shared_terminus_state"]
```

Comment:
This seems fine for now; if we ever do use this for another terminus, this design would give them all access to one another's state, so it might be nice if we could index into the appropriate terminus key in the map before we pass anything in to the terminus.  But there's a lot of speculation involved in that statement, so I think I'm fine with keeping it like this unless you have any ideas on that front that you think are worth the effort.